scripts/gis-material-update/process/modules/config.py

```diff
@@ -1,36 +1,111 @@
+from __future__ import annotations
+from os.path import exists
+from pathlib import Path
 import yaml
 
-from pathlib import Path
 
 class Config:
     """Class to handle configuration."""
 
     def __init__(self):
         self._cfg = self._cfg_file()
+        self._deployment_profile = "local_development"
+
+    def with_deployment_profile(self, deployment_profile: str) -> Config:
+        """Set deployment profile.
+
+        Supported configurations:
+            - local_development
+            - local_docker_development"""
+        self._deployment_profile = deployment_profile
+        return self
 
     def _cfg_file(self) -> dict:
-        """Try to find configuration file.
-        
+        """Find configuration file.
+
+        Return parsed configuration YAML.
+
         Handle cases
         * dockerized implementation
         * local development
         """
-        candidate_paths = [ "/haitaton-gis",  str(Path(__file__).parent.parent.parent) ]
-        for c_p in candidate_paths:
-            try:
-                with open("{}/config.yaml".format(c_p), "r") as stream:
+        parsed = {}
+        candidate_paths = [Path("/haitaton-gis"), Path(__file__).parent.parent.parent]
+
+        for config_file in [cand / "config.yaml" for cand in candidate_paths]:
+            if config_file.is_file():
+                with open(str(config_file), "r") as stream:
                     try:
                         parsed = yaml.safe_load(stream)
                     except yaml.YAMLError as exc:
                         raise Exception(exc)
-            except OSError as error:
-                print(f"Not found: {error.filename}")
-        return parsed
+                    else:
+                        print("Using configuration file: {}".format(config_file))
+                return parsed
+        raise OSError("Configuration file was not found.")
+
+    def _file_directory(self, storage: str = "download_dir") -> str:
+        """Obtain directory for input/output files.
+
+        Supported storage parameter values:
+            download_dir - directory for download files
+            output_dir - directory for gis output files."""
+        deployment_profile = self.deployment_profile()
+        directory = self._cfg.get("storage").get(deployment_profile, {}).get(storage)
+
+        if deployment_profile == "local_development":
+            directory_name = Path(__file__).parent.parent.parent / directory
+        elif deployment_profile == "local_docker_development":
+            directory_name = Path(directory)
+        else:
+            raise ValueError("Storage type not detected!")
```

Comment:
Is there any way around checking the environment here again? It feels like the config file should already define the directories.

Reply:
I am still struggling with this, how to (robustly) refer to directory in configuration file, several levels up.

I think I am keeping this implementation currently as-is, and will refactor it later if satisfactory solution is found.